test/k8sT/Services.go

```diff
@@ -1269,13 +1272,56 @@ var _ = Describe("K8sServicesTest", func() {
 			serverPort := data.Spec.Ports[1].TargetPort.IntValue()
 
 			// With ClusterIP
-			doFragmentedRequest(clientPod, srcPort, serverPort, data.Spec.ClusterIP, data.Spec.Ports[1].Port, true)
+			doFragmentedRequest(clientPod, srcPort, serverPort, data.Spec.ClusterIP, data.Spec.Ports[1].Port, testDS, true)
 
 			// From pod via node IPs
-			doFragmentedRequest(clientPod, srcPort+1, serverPort, k8s1IP, nodePort, hasDNAT)
-			doFragmentedRequest(clientPod, srcPort+2, serverPort, "::ffff:"+k8s1IP, nodePort, hasDNAT)
-			doFragmentedRequest(clientPod, srcPort+3, serverPort, k8s2IP, nodePort, hasDNAT)
-			doFragmentedRequest(clientPod, srcPort+4, serverPort, "::ffff:"+k8s2IP, nodePort, hasDNAT)
+			doFragmentedRequest(clientPod, srcPort+1, serverPort, k8s1IP, nodePort, testDS, hasDNAT)
+			doFragmentedRequest(clientPod, srcPort+2, serverPort, "::ffff:"+k8s1IP, nodePort, testDS, hasDNAT)
+			doFragmentedRequest(clientPod, srcPort+3, serverPort, k8s2IP, nodePort, testDS, hasDNAT)
+			doFragmentedRequest(clientPod, srcPort+4, serverPort, "::ffff:"+k8s2IP, nodePort, testDS, hasDNAT)
+		}
+
+		testIPv4FragmentMetrics := func() {
+			var (
+				data    v1.Service
+				srcPort = 12345
+				hasDNAT = true
+			)
+
+			ciliumPodK8s2, err := kubectl.GetCiliumPodOnNode(helpers.K8s2)
+			ExpectWithOffset(2, err).Should(BeNil(), "Cannot get cilium pod on k8s2")
+
+			// Destination address and port for fragmented datagram
+			// are not DNAT-ed with kube-proxy but without bpf_sock.
+			if helpers.RunsWithKubeProxy() {
+				hasDNAT = kubectl.HasHostReachableServices(ciliumPodK8s2, false, true)
+			}
+
+			clientPodK8s1, _ := kubectl.GetPodOnNodeLabeledWithOffset(helpers.K8s1, testDSClient, 1)
+
+			// Pick test-nodeport-local-k8s2 as service/destination
+			// port, which has a `Local` `externalTrafficPolicy`.
+			//
+			// This will ensure packets hitting k8s2 will not be
+			// fowarded back to the k8s1 node.
+			err = kubectl.Get(helpers.DefaultNamespace, "service test-nodeport-local-k8s2").Unmarshal(&data)
+			ExpectWithOffset(1, err).Should(BeNil(), "Cannot retrieve service")
+			nodePort := data.Spec.Ports[1].NodePort
+			serverPort := data.Spec.Ports[1].TargetPort.IntValue()
+
+			cmd := "cilium bpf metrics list | grep 'Fragmented packet' | grep INGRESS | awk '{print $4}'"
```

Comment:
Nit: Since you invoke awk already, you might as well avoid spawning grep twice? Something like (not tested):
```suggestion
			cmd := "cilium bpf metrics list | awk '/Fragmented packet.*INGRESS/ {print $4}'"
```
Ideally, we would even go for `-o jsonpath={...}`, but I had a quick look at the JSON output for that command and it looks... original. Let's stick plain output + awk for now, I guess.

Reply:
Nice, I should start learning awk for real :) 

> (not tested

(tested, works fine)